pkg/controllers/node/termination/terminator/events/events.go

```diff
@@ -34,6 +34,16 @@ func EvictPod(pod *v1.Pod) events.Event {
 	}
 }
 
+func DeletePod(pod *v1.Pod) events.Event {
+	return events.Event{
```

Comment:
IMO, we should make an events.k8s.io/v1 [Event](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/) with:
- the Pod being marked for deletion as the `regarding` object
- the NodePool [for the NodeClaim for the Node where that Pod is running] as the `related` object

K8s' API lets us identify two objects, and there are two relevant things, so let's mention both

Reply:
I do not see a way to reference a related object using the client-go EventRecorder.
https://pkg.go.dev/k8s.io/client-go/tools/record#EventRecorder

Reply:
I'm not sure what the benefit of creating the event directly here is. We should use the existing eventing solution we have in the codebase rather than creating a new mechanism.

Reply:
recorder.Event() is the official documented way to use client-go to produce k8s Events in the API.
https://pkg.go.dev/k8s.io/client-go/tools/record

Karpenter does have a centralized interface that could potentially be updated to bypass the expected interface by creating Event resources directly if we thought it would be worth it to do so.
https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/events/recorder.go

I do agree that it should be handled in a separate event refactoring PR if desired.